src/bootstrap/src/core/config/config.rs

```diff
@@ -2397,6 +2397,15 @@ impl Config {
             );
         }
 
+        let has_external_llvm_config = !config.llvm_from_ci
```

Comment:
Nit: We could move `Builder::is_system_llvm` to `Config` since it doesn't rely on any `Builder` specific data (other than `is_builder_target` function which can also be moved).